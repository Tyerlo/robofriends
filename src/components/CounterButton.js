import React, { Component } from "react";

class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0
    };
  }
  shouldComponentUpdate(nextProps, nextState) {
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }

  updateCount = () => {
    const { count } = this.state;
    this.setState({
      count: count + 1
    });
  };
  render() {
    const { count } = this.state;

    return (
      <button id="counter" color={"red"} onClick={this.updateCount}>
        Count: {count}
      </button>
    );
  }
}
export default CounterButton;
