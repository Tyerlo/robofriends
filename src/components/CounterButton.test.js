import React from "react";
import { shallow } from "enzyme";
import CounterButton from "./CounterButton";

it("expect to render counter component", () => {
  const mockColor = "red";
  expect(shallow(<CounterButton color={mockColor} />)).toMatchSnapshot();
});

it("correctly increments the counter", () => {
  const wrapper = shallow(<CounterButton />);
  wrapper.find('[id="counter"]').simulate("click");
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 2 });

  expect(wrapper.props().color).toEqual("red");
});
