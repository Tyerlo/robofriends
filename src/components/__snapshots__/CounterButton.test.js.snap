// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render counter component 1`] = `ShallowWrapper {}`;
